client/api/src/light.rs

```diff
@@ -90,6 +90,29 @@ pub struct RemoteReadChildRequest<Header: HeaderT> {
 	pub retry_count: Option<usize>,
 }
 
+/// Remote storage range read request.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct RemoteReadRangeRequest<Header: HeaderT> {
+	/// Read at state of given block.
+	pub block: Header::Hash,
+	/// Header of block at which read is performed.
+	pub header: Header,
+	/// Storage key of child trie to query, `None` for top trie.
+	pub child_trie_key: Option<PrefixedStorageKey>,
+	/// Limit range of query to a given prefix.
+	pub prefix: Option<Vec<u8>>,
+	/// Maximum number of key value to query.
+	pub count: u32,
+	/// Maximum total size of value to query.
+	/// Acts as a threshold.
+	pub value_size: u32,
```

Comment:
I'd suggest this to be called `max_value_size` or `value_size_limit` here and everywhere.